src/cli/bin.js

```diff
@@ -3,79 +3,65 @@
 /* eslint-disable no-console */
 'use strict'
 
-process.on('uncaughtException', (err) => {
-  console.info(err)
-
-  throw err
-})
-
-process.on('unhandledRejection', (err) => {
-  console.info(err)
-
-  throw err
-})
-
 const semver = require('semver')
-const pkg = require('../../package.json')
-
-if (!semver.satisfies(process.versions.node, pkg.engines.node)) {
-  console.error(`Please update your Node.js version to ${pkg.engines.node}`)
-  process.exit(1)
-}
-
 const YargsPromise = require('yargs-promise')
 const updateNotifier = require('update-notifier')
-const { print } = require('./utils')
-const mfs = require('ipfs-mfs/cli')
 const debug = require('debug')('ipfs:cli')
 const parser = require('./parser')
 const commandAlias = require('./command-alias')
+const { print } = require('./utils')
+const pkg = require('../../package.json')
 
-function main (args) {
-  const oneWeek = 1000 * 60 * 60 * 24 * 7
-  updateNotifier({ pkg, updateCheckInterval: oneWeek }).notify()
+// Handle any uncaught error
+process.once('uncaughtException', (err, origin) => {
+  if (origin === 'uncaughtException') {
+    print(err.message)
+    debug(err)
+  }
+})
+process.once('unhandledRejection', (err) => {
+  print(err.message)
+  debug(err)
+})
 
-  const cli = new YargsPromise(parser)
+// Check for node version
+if (!semver.satisfies(process.versions.node, pkg.engines.node)) {
+  console.error(`Please update your Node.js version to ${pkg.engines.node}`)
+  process.exit(1)
+}
```

Comment:
The `uncaughtException`/`unhandledRejection` handlers and node version check need to be above the `require` statements as they were because any one of them could cause an error or use code that is not supported in the version of Node.js that the user is running.